.github/workflows/android_builds.yml

```diff
@@ -0,0 +1,52 @@
+on:
+  pull_request:
+    branches:
+      - main
+    # types:
+    #   - closed
+
+name: Update Android Artifacts
+
+permissions:
+  contents: write
```

Comment:
Maybe prudent to leave a warning at the top of this YML that changes to this workflow should be carefully vetted since it can write to the repository.

I wonder if branch protection rules to require pull-requests into `main` would protect against this workflow pushing `main` or if that would be possible through malice or bugs in the script :-/

Reply:
I believe branch protection rules do apply to Workflow Actions as well, although I'm unsure we can exactly test the existing `main` protection rule without risking a little bit.

Following this comment from you:
> If there's a way to use branch protection rules (or some other mechanism) to further limit the workflow to only being able to operate with the Android maven branches we intend that would be a nice extra addition 🤞

I tried adding a custom ruleset (see [here](https://github.com/rustls/rustls-platform-verifier/settings/rules/88258)) to the repository that applies to every branch except `main-with-maven`. I then tried to push from the CI job to a testing branch. [It failed to push](https://github.com/rustls/rustls-platform-verifier/actions/runs/6132043096/job/16642549893?pr=33) with a permissions error like we'd want.

The downside of it seems to be that it treats our pushes to working branches in the repository as rule bypasses as well, which causes extra noise locally 😞
```
remote: Bypassed rule violations for refs/heads/automated-android-branch:
remote:
remote: - Cannot update this protected ref.
remote:
To github.com:rustls/rustls-platform-verifier.git
   036ac7e..e147cb2  automated-android-branch -> automated-android-branch
```

Reply:
> The downside of it seems to be that it treats our pushes to working branches in the repository as rule bypasses as well, which causes extra noise locally

Interesting! Thanks for testing.

I'm in the habit of pushing branches to a fork and opening PRs from there, so I don't think the noise would bother me but I can see how its annoying for others that don't want the overhead of managing branches on a fork.

Reply:
To clarify, I am personally fine with the noise since I know what it means, and the gained benefit of locking this job down tightly feels very worth it. If I ever get tired of it I could also switch to using a fork. IMO given that we should keep the rule before merging this and live with the very minor downsides.

Reply:
We could also consider adding a CODEOWNERS rule to this file to require another approval to prevent accidentally (though not malicious) updates to this workflow that are incorrect and have security consequences.

Reply:
> We could also consider adding a CODEOWNERS rule to this file to require another approval

I wouldn't mind that extra caution if you wouldn't :-)